packages/jupyter-chat/src/model.ts

```diff
@@ -648,3 +673,32 @@ export namespace IChatModel {
     documentManager?: IDocumentManager | null;
   }
 }
+
+/**
+ * Interface of the chat context, a 'subset' of the model with readonly attribute,
+ * which can be passed to the input model.
+ * This allows third party extensions to get some attribute of the model without
+ * exposing the method that can modify it.
+ */
+export interface IChatContext {
+  /**
+   * The name of the chat.
+   */
+  readonly name: string;
+}
+
+/**
+ * A default implementation of the IChatContext, that can be extended to provide more
+ * attributes.
+ */
+export class ChatContext implements IChatContext {
+  constructor(options: { model: IChatModel }) {
+    this._model = options.model;
+  }
+
+  get name(): string {
+    return this._model.name;
+  }
+
+  protected _model: IChatModel;
+}
```

Comment:
Forgot to add this to my review.

Just curious, what is the reasoning behind have a `ChatContext` and a `LabChatContext`? It may not be clear which one is the "main" implementation used in Jupyter Chat. Instead, it could be better to stick to one, and let developers provide their own implementation if they want to use another. What do you think?

Reply:
The `ChatContext` is the default implementation of the context, while `LabChatContext` is extending it to use it in `jupyterlab-chat`.